internal/backends/dart/dart.go

```diff
@@ -0,0 +1,250 @@
+		util.Die("pubspec.lock: %s", err)
+	}
+	pkgs := map[api.PkgName]api.PkgVersion{}
+	for nameStr, data := range cfg.Packages {
+		pkgs[api.PkgName(nameStr)] = api.PkgVersion(data.Version)
+	}
+	return pkgs
+}
+
+// pubDevSearchResults represents the data we get from Pub.dev when
+// doing a search.
+type pubDevSearchResults struct {
+	Packages []struct {
+		Name string `json:"package"`
+	} `json:"packages"`
+	NextURL string `json:"next"`
+}
+
+// dartSearch implements Search for Pub.dev.
+func dartSearch(query string) []api.PkgInfo {
+	endpoint := "https://pub.dev/api/search"
+	queryParams := "?q=" + url.QueryEscape(query)
+
+	resp, err := http.Get(endpoint + queryParams)
+	if err != nil {
+		util.Die("Pub.dev: %s", err)
+	}
+	defer resp.Body.Close()
+
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		util.Die("Pub.dev: %s", err)
+	}
+
+	var pubDevResults pubDevSearchResults
+	if err := json.Unmarshal(body, &pubDevResults); err != nil {
+		util.Die("Pub.dev: %s", err)
+	}
+
+	results := make([]api.PkgInfo, len(pubDevResults.Packages))
+	for i := range pubDevResults.Packages {
+		p := pubDevResults.Packages[i]
+		results[i] = api.PkgInfo{
+			Name: p.Name,
+		}
+	}
+	return results
+}
+
+// dartInfo is an empty stub implementation.
+func dartInfo(name api.PkgName) api.PkgInfo {
+	return api.PkgInfo{
+		Name:          "",
+		Description:   "",
+		Version:       "",
+		HomepageURL:   "",
+		SourceCodeURL: "",
+		BugTrackerURL: "",
+		Author: util.AuthorInfo{
+			Name:  "",
+			Email: "",
+			URL:   "",
+		}.String(),
+		License: ""}
+}
+
+func createSpecFile() {
+	if util.Exists("pubspec.yaml") {
+		util.Die("pubspec.yaml already exists")
+	}
+
+	pubspec := dartPubspecYaml{
+		Name: "MyApp",
+	}
+	writeSpecFile(pubspec)
+}
+
+func writeSpecFile(specs dartPubspecYaml) {
+	var data []byte
+	var err error
+	data, err = yaml.Marshal(&specs)
+	if err != nil {
+		fmt.Println("Marshal Error")
+	}
+
+	ioutil.WriteFile("pubspec.yaml", data, 0666)
+}
+
+func readSpecFile() dartPubspecYaml {
+	contentsB, err := ioutil.ReadFile("pubspec.yaml")
+	if err != nil {
+		util.Die("pubspec.yaml: %s", err)
+	}
+
+	var specs dartPubspecYaml
+	if err := yaml.Unmarshal(contentsB, &specs); err != nil {
+		util.Die("pubspec.yaml: %s", err)
+	}
+
+	return specs
+}
+
+func dartAdd(pkgs map[api.PkgName]api.PkgSpec) {
+	if !util.Exists("pubspec.yaml") {
+		createSpecFile()
+	}
+
+	var specs dartPubspecYaml
+	specs = readSpecFile()
+
+	for name, spec := range pkgs {
+		arg := string(name)
+		if spec != "" {
+			specs.Dependencies[arg] = string(spec)
+		} else {
+			specs.Dependencies[arg] = nil
+		}
+	}
+
+	writeSpecFile(specs)
+}
+
+func dartRemove(pkgs map[api.PkgName]bool) {
+	var specs dartPubspecYaml
+	specs = readSpecFile()
+
+	for name := range pkgs {
+		delete(specs.Dependencies, string(name))
+	}
+
+	writeSpecFile(specs)
+}
+
+// dartGuess stub.
+func dartGuess() (map[api.PkgName]bool, bool) {
+	pkgs := map[api.PkgName]bool{}
+
+	return pkgs, true
+}
```

Comment:
It's fine to panic with NotImplemented for this, I think. The fact that it isn't implemented can be documented in the README.

Reply:
Done